docs/content/dev/authentication.md

```diff
@@ -0,0 +1,87 @@
+# User Profiles and Authentication
+Each user in HedgeDoc 2 has a profile
+which contains the following information:
+
+- username (`janedoe`)
+- display name (`Jane Doe`)
+- email address, optional (`janedoe@example.com`)
+- profile picture, optional
+- the date the user was created
+- the date the profile was last updated 
+
+HedgeDoc 2 supports multiple authentication methods per user.  
+These are called *identities* and each identity is backed by an
+auth provider (like OAuth, SAML, LDAP or internal auth).
+
+One of a users identities may be marked as *sync source*.  
+This identity is used to automatically update profile attributes like the
+display name or profile picture on every login. If a sync source exists, the
+user can not manually edit their profile information.
+If an external provider was used to create the account,
+it is automatically used as sync source. 
+
+The administrator may globally set one or more auth providers as sync source,
+e.g. to enforce that all profile information comes from the corporate
+LDAP and is the same across multiple applications.  
+If global sync sources exist, new accounts can only be created using
+these auth providers. The auth provider that was used to create the account
+is automatically set as sync source and cannot be changed by the user.
+This effectively pins the account to this provider.
+
+## Example: Corporate LDAP
+The administrator wants to allow users to log in via the corporate LDAP
+and Google. Login must only be possible for users present in LDAP and 
+all users must be displayed as they are in the LDAP.
+
+The admin therefore sets up two login providers:
+
+- corporate LDAP, marked as global sync source
+- Google OAuth login
+
+If a new user tries to log in via Google, they will not be found in the
+database. The frontend detects that a global sync source exists and
+suggests logging in via LDAP first.
+
+After a new user created their account by logging in via LDAP, they can use
+the 'add a new login method' feature in their profile page to link their
+Google account and use it to login afterwards.
+
+## Example: Username Conflict
+HedgeDoc is configured with two auth providers.
+
+- A user logs in using auth provider A.
+- The backend recieves the profile information from provider A and notices that the username in the profile
+  already exists in the database, but no identity for this provider-username combination exists.
+- The backend creates a new user with another username to solve the username conflict.
```

Comment:
How is this other username generated?

Reply:
That is an implementation detail and intentionally not specified